### PR TITLE
Fix ImportantFileWriterTest.

### PR DESCRIPTION
**Pull request by ghost**
b/316401416

### CODE REVIEW

File: base/files/file_util.cc [hunk 1]
```diff
@@ -369,7 +369,45 @@ bool ReadFileToStringWithMaxSize(const FilePath& path,
     return false;
   return ReadStreamToStringWithMaxSize(file_stream.get(), max_size, contents);
 #endif
+#if defined(STARBOARD)
```
Comment (andrewsavage1) [inline base/files/file_util.cc:372]: Let's make this part of the above macro so we know we'd like to clean it up

Also I know @sherryzy was looking at implementing the above types already so the upstream code worked as is, did anything come from that?

File: base/files/important_file_writer_unittest.cc [hunk 2]
```diff
@@ -367,6 +367,7 @@ TEST_F(ImportantFileWriterTest, DoScheduledWrite_FailToSerialize) {
   histogram_tester.ExpectTotalCount("ImportantFile.WriteDuration", 0);
 }
 
+#if !defined(STARBOARD)
```
Comment (andrewsavage1) [inline base/files/important_file_writer_unittest.cc:370]: Are the disabled tests inherently incompatible with cobalt or do we want to enable them later on?

File: base/files/file_util.cc (hunk 1 above)
Comment (andrewsavage1) [reply base/files/file_util.cc]: I agree with that, but if Sherry has already gotten the upstream code working, we can just use that. If not, then let's use this

File: base/files/file_util.cc [hunk 3]
```diff
@@ -363,13 +363,48 @@ bool ReadFileToStringWithMaxSize(const FilePath& path,
...
+  read_status = read_status && file.IsValid();
+  return read_status;
+#else
   ScopedFILE file_stream(OpenFile(path, "rb"));
   if (!file_stream)
     return false;
   return ReadStreamToStringWithMaxSize(file_stream.get(), max_size, contents);
-#endif
-  return false;
+#endif  // defined(USE_HACKY_COBALT_CHANGES)
```
Comment (sherryzy) [inline base/files/file_util.cc:407]: When I'm fixing JSONReaderTest, I did the same fix here. Andrew left a comment about try to implement upstream code, see https://github.com/youtube/cobalt/pull/2084/files/e4306067ff231d5ee59a66c7cd58058ca5957320#r1426999564.

I didn't really look into this yet. Would you like to take a look?

**Review by sherryzy — APPROVED** [review]

**Pull request merged**